packages/http-client-java/emitter/src/versioning-utils.ts

```diff
@@ -0,0 +1,57 @@
+import { SdkClientType, SdkHttpOperation } from "@azure-tools/typespec-client-generator-core";
+import { Namespace, Program } from "@typespec/compiler";
+import { findVersionedNamespace, getVersions } from "@typespec/versioning";
+
+/**
+ * Gets the array of api-version on the TypeSpec service that contains this SDK client.
+ * `undefined` if the service is not versioned.
+ *
+ * @param program the program
+ * @param client the SDK client
+ * @returns the array of api-version on the TypeSpec service that contains this SDK client
+ */
+export function getServiceApiVersions(
+  program: Program,
+  client: SdkClientType<SdkHttpOperation>,
+): string[] | undefined {
+  // TODO: use client.apiVersions after TCGC supports multiple service
+  let apiVersions: string[] | undefined;
+  const verVersionedNamespace: Namespace | undefined = findVersionedNamespace(
+    program,
+    client.__raw.service,
+  );
+  if (verVersionedNamespace) {
+    apiVersions = getVersions(program, verVersionedNamespace)[1]
+      ?.getVersions()
+      .map((version) => version.value);
+  }
+  return apiVersions;
+}
+
+/**
+ * Filter api-versions for "ServiceVersion".
+ * TODO(xiaofei) pending TCGC design: https://github.com/Azure/typespec-azure/issues/965
```

Comment:
We will continue to use this function. As we use `getServiceApiVersions` above, we need this function to filer-out "newer" api-version than specified.